### DISCUSSION TITLE
Feature Request: Syntax highlighting in diff code fences

### DISCUSSION THREAD

**Discussion by Radiergummi** [community/community#62038] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

In places GitHub accepts Markdown, you can either have highlighted code blocks:
```js
function helloWord() {
  return 'Hi, Github!';
}
```

or you can have diff code blocks to highlight changes:
```diff
function helloWord() {
- return 'Hi, Github!';
+ return 'Hi, GitHub!';
}
```

but never both. As you can see, using the code block annotated with `diff` looses all highlighting related to the language, keeping only the additions and removals. This works in simple cases, but makes larger code snippets straining to read.  

I'd like to propose adding `+diff` as a modifier to any other language, like so:
<pre>
```js+diff
function helloWord() {
- return 'Hi, Github!';
+ return 'Hi, GitHub!';
}
```
</pre>

Where both the JavaScript highlighter _and_ the diff highlighter would process the code.

**Comment by CodeMaster7000** [comment]
Great idea!
